webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/JavascriptEngine.scala

```diff
@@ -612,35 +615,42 @@ final object JsEngine {
       }
 
       try {
-        // submit to the pool and synchroniously retrieve the value with a timeout
-        pool.submit(scriptCallable).get(maxTime._1.toLong, maxTime._2)
+        val task = pool.submit(scriptCallable)
+        try {
+          // submit to the pool and synchroniously retrieve the value with a timeout
+          task.get(maxTime.toMillis, TimeUnit.MILLISECONDS)
       } catch {
         case ex: ExecutionException => //this can happen when rhino get security exception... Yeah...
           throw RudderFatalScriptException(s"Evaluating script '${name}' was forced interrupted due to ${ex.getMessage}, aborting.", ex)
 
         case ex: TimeoutException =>
           //try to interrupt the thread
           try {
-            // try to gently terminate the thread
-            pool.shutdownNow()
```

Comment:
What this throwing the 'threadpool' exception we used to get sometimes ? would be logic though, threadpool was shutdown but other thread could still be created/using the threadpool

Reply:
exactly